hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -996,77 +996,289 @@
     }
   ],
 
-  // ===== INTERRUPT CTRL =====================================================
-
-  // RV_PLIC has two searching algorithm internally to pick the most highest priority interrupt
-  // source. "sequential" is smaller but slower, "matrix" is larger but faster.
-  // Choose depends on the criteria. Currently it is set to "matrix" to meet FPGA timing @ 50MHz
-
-  // ===== ALERT HANDLER ======================================================
+  // ===== PINMUX & PINOUT ======================================================
+
+  pinout: {
+    // IO power bank declaration.
+    // This list defines the IO bank power domains on the ASIC.
+    // Each individual pad must be in one of the declared power domains.
+    banks: ['VCC', 'AVCC', 'VIOA', 'VIOB'],
+    // Pad declaration.
+    // Each entry must have the following four keys:
+    //
+    // - name: Name of the pad (this will be exposed at the chiplevel).
+    //
+    // - type: Pad type (this maps to the pad types defined in prim_pad_wrapper_pkg.sv)
+    //
+    // - bank: Specifies in which of the IO power banks this pad lives.
+    //
+    // - muxed: If this is set to True, the pad will be connected to the pinmux matrix.
+    //          If this is set to False, the pad can be referenced by direct connections in
+    //          the pinmux signal mapping further below.
+    //          If an unmuxed pad is NOT referenced in the pinmux signal mapping,
+    //          the pad needs to be wired up manually in the template.
```

Comment:
i think i get what you're going for here, but it makes me wonder if the "implied" manual connection pad instead of explicit could bite us in unexpected ways.

I think what you're saying is "if something is marked as a dedicated pad, yet there is no specific connection to it, it must therefore be a manual connection".  is that right?

Reply:
Yes that is correct.
The template generates the signal definitions and padring connections in the toplevel, and lint would flag that these are not connected.
But I see what you are implying. 
Should we replace the "muxed" key with something similar as the "connection" key in the pinmux config below?

Reply:
i'm not sure.... let me think a bit. Because based on what we are discussing, i don't really see an escape case.
do you?

Reply:
With the current tooling in place, we should be able to detect this - all tools parsing the output should report an error.
That being said, making it explicit would allow us to catch these errors already during topgen validation/merge.
Let me quickly check how difficult it would be to change this.

Reply:
Ok I have made this more explicit by adding a `connection` type to the pads as well.
This also allows for more rigorous error checking during the validation phase.